python/lsst/sconsUtils/state.py

```diff
@@ -172,14 +172,20 @@ def _initEnvironment():
     if env['PLATFORM'] == 'darwin':
         env['LDMODULESUFFIX'] = ".so"
         if not re.search(r"-install_name", str(env['SHLINKFLAGS'])):
-            env.Append(SHLINKFLAGS=["-Wl,-install_name", "-Wl,${TARGET.file}"])
+            env.Append(SHLINKFLAGS=["-install_name", "@rpath/${TARGET.file}"])
```

Comment:
As you say, adding `@rpath` is a no-op when you set `DYLD_LIBRARY_PATH` as the linker always searches that path first regardless of any path burned into the library. I don't quite understand what you are saying about proper `$PREFIX` because we don't currently set any prefix at all in the install name and furthermore `conda-build` explicitly goes in and changes all our library install names to use `@rpath`. From reading the notes about `@rpath` I was under the impression that adding `@rpath` was the only way to allow people to reliably use your libraries if they aren't setting `DYLD_LIBRARY_PATH` because they can set the search path in the binaries explicitly and indirect dependent libraries will know to also use that search path. Most of the packages outside of EUPS do already do what this patch does (see e.g. boost).